core/src/main/java/org/bitcoinj/crypto/ChildNumber.java

```diff
@@ -73,9 +73,10 @@ public ChildNumber(int i) {
      */
     public static ChildNumber parse(String str) {
         boolean isHard = str.endsWith("H");
-        if (isHard) str = str.substring(0, str.length() - 1).trim();
-        int nodeNumber = Integer.parseInt(str);
-        return new ChildNumber(nodeNumber, isHard);
+        String nodeNumber = (isHard)
+                ? str.substring(0, str.length() - 1)
+                : str;
+        return new ChildNumber(Integer.parseInt(nodeNumber.trim()), isHard);
```

Comment:
I just analyzed for a while why you changed the logic around `trim()` and then noticed the somewhat hidden comment about the bugfix.

I think if we encounter such bugs, we should fix them on a separate commit/PR, and preferable directly based on master so that the chance of a clean backport is as high as possible.

Anyway, I split this commit into two commits and merged.

Reply:
Yeah, I should have used a separate commit or at least mentioned the fix in the 1-line summary.